src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -455,19 +455,18 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
 
             if (hasExplicitNames)
             {
-                // If the tuple type with names is bound in a declaration
-                // context then we must have the TupleElementNamesAttribute to emit
-                if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute())
-                {
-                    // Report diagnostics if System.String doesn't exist
-                    this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+                // If the tuple type with names is bound we must have the TupleElementNamesAttribute to emit
+                // it is typically there though, if we have ValueTuple at all
+                // and we need System.String as well
 
-                    if (!Compilation.HasTupleNamesAttributes)
-                    {
-                        var info = new CSDiagnosticInfo(ErrorCode.ERR_TupleElementNamesAttributeMissing,
-                            AttributeDescription.TupleElementNamesAttribute.FullName);
-                        Error(diagnostics, info, syntax);
-                    }
+                // Report diagnostics if System.String doesn't exist
+                this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+
+                if (!Compilation.HasTupleNamesAttributes)
```

Comment:
That is the current idea. 

The general expectation here is that the situations where tuple types are available and the attribute is not, will be nearly impossible, so investing in a better diagnostics here is not very valuable.


---
In reply to: [80978929](https://github.com/dotnet/roslyn/pull/14023#discussion_r80978929) [](ancestors = 80978929)